lib/src/rules/equals_test_argument_type.dart

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.equals_test_argument_type;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/linter.dart';
+
+const desc = r'"operator ==(Object other)" should test argument type.';
+
+const details = r'''
+**AVOID** "operator ==(Object other)" should test argument type.
```

Comment:
How about something like 

**ALWAYS** test type arguments in `operator ==(Object other)`.